### ISSUE TITLE
Change `REAME.md` template context `included_projects` to `projects`

### ISSUE THREAD

**Issue by nozomione** [AlexsLemonade/scpca-portal#781]
### Context

**Updated:** The requirements for this issue have been adjusted to account for upcoming changes. See https://github.com/AlexsLemonade/scpca-portal/issues/781#issuecomment-2236916471.

Related issue: [Additional Restrictions for Portal-wide metadata download](https://github.com/AlexsLemonade/ScPCA-admin/issues/908)

Additional restrictions may apply to the data contained in each project, and we want to include this information in the metadata-only downloads' `README.md` file.

<details>
   <summary>Additional restrictions for the project <code>SCPCP000001</code></summary><br/>

![Screenshot 2024-06-27 at 4 48 09 PM](https://github.com/nozomione/ccdl/assets/31800566/e9a7355e-e52f-495b-8bd4-6a1fd90b6af1)
</details>

Currently, the content of additional restrictions for a single project is rendered in a downloadable `README.md` file as follows:

```markdown
### Additional Restrictions
This dataset is designated as research or academic purposes only.
```
Since the same README template file (`templates/readme/metadata_only.md`) will be used for both project-level and portal-wide metadata-only downloads and the latter must support multiple projects, we’ll need to update this template to enable the display of the following table with the project ID and its corresponding data usage restrictions.

A table with multiple project information:
|Project ID|Data Usage Restrictions|
|:--|:--|
|SCPCP000001|Research or academic purposes only.|
|SCPCP000002|Research or academic purposes only.|
|SCPCP000003|Research or academic purposes only.|

### Problem or idea
In order to generate the table above in the shared README template, we should pass an array of projects as the template context instead of `self.ATTRIBUTE_NAME`.

### Solution or next step
1. Update `Project::create_metadata_readme_file` as follows:
- Rename the context variable name `included_projects` to `projects` 
- Pass an array of `self` (a project) as the template context, not just specific attributes
 
 ```py
 def create_metadata_readme_file(self):
      """Creates a metadata only README file."""

        with open(ComputedFile.README_METADATA_PATH, "w") as readme_file:
            readme_file.write(
                render_to_string(
                    ComputedFile.README_TEMPLATE_METADATA_PATH,
                    context={
                        "additional_terms": self.get_additional_terms(),
                        "date": utils.get_today_string(),
                        "projects": [self],  # here
                    },
                ).strip()
            )
```

2. Update the context variable name `included_projects` to `projects` in`templates/readme/metadata_only.md` 

e.g.) 
```md
 ## Contents
 {% if projects|length > 1 %}
 This download includes associated metadata for samples from all projects currently available at time of download in the ScPCA portal.
 {% else %}
 {% for project_accession, project_url in projects %}
 This download includes associated metadata for samples from project [{{ project_accession }}]({{ project_url }}) in the ScPCA portal.
 {% endfor %}
 {% endif %}
 ```

**Comment by nozomione** [comment]
> So for this issue we just want to address the existing metadata readme template. Looks like you wrote it for the portal wide.

Oh based on the discussion [2024-06-25 - ScPCA Metadata](https://data-lab-knowledge.slab.com/posts/2024-06-25-sc-pca-metadata-1gb200oo), I thought we'll be writing the issue for the portal-wide. Yes, I'll update that for project-level 👍 !

**Comment by nozomione** [comment]
Now, we'll be making changes the template context for all create-readme methods (not just metadata-only) and the context variable names of their corresponding readme templates will be adjusted as well. 

The following methods and readme templates will be updated:
- `create_anndata_readme_file` and `anndata.md`
- `create_anndata_merged_readme_file` and `anndata_merged.md`
- `create_metadata_readme_file` and `metadata_only.md`
- `create_multiplexed_readme_file` and `multiplexed.md`
- `create_single_cell_readme_file` and `single_cell.md`
- `create_single_cell_merged_readme_file` and `single_cell_merged.md`
- `create_spatial_readme_file` and `spatial.md`

**Comment by davidsmejia** [comment]
closed with #790
